### PR TITLE
Optimize graphql fragments repository

### PR DESCRIPTION
**Pull request by zzunebye**
1. This branch reduces the blocs' dependencies over data layer, mainly by injecting the repositories through constructor outside the bloc class. -> (View - Bloc - Repository - Service - API)
2. reduce the duplicated code by using fragments in GraphQL queries.
3. folder structure is tidy up, deleting legacy files and changes the confusing name.

### CODE REVIEW

File: lib/pages/job_search_page/job_search_page.dart [hunk 1]
```diff
@@ -25,13 +25,13 @@ class _JobSearchPageState extends State<JobSearchPage> {
...
     super.initState();
-    _searchBloc = SearchBloc();
+    _searchBloc = BlocProvider.of<SearchBloc>(context);
   }
 
   @override
   void dispose() {
     super.dispose();
-    _searchBloc.close();
+    // _searchBloc.close();
```
Comment (kenho85) [inline lib/pages/job_search_page/job_search_page.dart]: remove?

File: lib/repositories/job_repository.dart [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+class PostRepository {
+  final IJobService _dataService;
+
+  PostRepository(this._dataService) ;
+
+  getJobPosts(int limit) {
+    try {
+      final result = _dataService.getJobPosts(limit);
+
```
Comment (kenho85) [inline lib/repositories/job_repository.dart]: empty space

File: lib/pages/job_search_page/job_search_page.dart (hunk 1 above)
Comment (zzunebye) [reply lib/pages/job_search_page/job_search_page.dart]: Removed. Thanks

File: lib/repositories/job_repository.dart (hunk 2 above)
Comment (zzunebye) [reply lib/repositories/job_repository.dart]: I changed to directly return the results without declaring new variables

File: lib/services/hive_service.dart [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    await Hive.openBox('seekerPrefBox');
+  }
+
+  get() {
+    try {
+      final _prefBox = Hive.box('seekerPrefBox');
+
+      if (_prefBox.isEmpty || !_prefBox.containsKey('myPref')) _prefBox.put('myPref', emptyPrefValue);
+
+      return Hive.box('seekerPrefBox').get('myPref').cast<Preference>();
```
Comment (kenho85) [inline lib/services/hive_service.dart]: return _prefBox.get('myPref').cast<Preference>();

**Review by kenho85 — APPROVED** [review]

**Pull request merged**